pkg/route/ingress/ingress.go

```diff
@@ -423,6 +423,9 @@ func (c *Controller) sync(key queueKey) error {
 			continue
 		}
 		for _, path := range rule.HTTP.Paths {
+			if ingress.Annotations["route.openshift.io/termination"] == "passthrough" {
+				path.Path = ""
```

Comment:
Don't mutate the ingress; define and use new variable `path`, similar to what you did here:  https://github.com/openshift/openshift-controller-manager/pull/169/files#diff-5100b0e98310cb270f5b9b2bc09313e46cdb9c509eef0ea3a76dda2a07ce987bR426-R431